src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -522,32 +522,50 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 			return shell.openExternal(url);
 		}
 
+		let browserName = configuredBrowser;
+		if (configuredBrowser.toLowerCase() === 'edge') {
+			browserName = isWindows ? 'msedge' : 'microsoft-edge';
+		}
+
 		if (configuredBrowser.includes(posix.sep) || configuredBrowser.includes(win32.sep)) {
 			const browserPathExists = await Promises.exists(configuredBrowser);
 			if (!browserPathExists) {
 				this.logService.error(`Configured external browser path does not exist: ${configuredBrowser}`);
-				return shell.openExternal(url);
+				return;
 			}
 		}
 
 		try {
 			const { default: open } = await import('open');
 			const res = await open(url, {
 				app: {
-					// Use `open.apps` helper to allow cross-platform browser
-					// aliases to be looked up properly. Fallback to the
-					// configured value if not found.
-					name: Object.hasOwn(open.apps, configuredBrowser) ? open.apps[(configuredBrowser as keyof typeof open['apps'])] : configuredBrowser
-				}
+					name: browserName
+				},
+				wait: true
 			});
 
+			if (res.exitCode !== 0) {
+				throw new Error(`Failed to open browser. Exit code: ${res.exitCode}`);
+			}
+
 			res.stderr?.once('data', (data: Buffer) => {
-				this.logService.error(`Error openening external URL '${url}' using browser '${configuredBrowser}': ${data.toString()}`);
-				return shell.openExternal(url);
+				this.logService.error(`Error opening external URL '${url}' using browser '${browserName}': ${data.toString()}`);
 			});
 		} catch (error) {
-			this.logService.error(`Unable to open external URL '${url}' using browser '${configuredBrowser}' due to ${error}.`);
-			return shell.openExternal(url);
+			this.logService.error(`Unable to open external URL '${url}' using browser '${browserName}' due to ${error}.`);
+			try {
+				let command;
+				if (isWindows) {
+					command = `start ${browserName} "${url}"`;
+				} else if (isMacintosh) {
+					command = `open -a "${browserName}" "${url}"`;
+				} else {
+					command = `${browserName} "${url}"`;
+				}
+				await promisify(exec)(command);
+			} catch (fallbackError) {
+				this.logService.error(`Fallback method failed to open URL '${url}' using browser '${browserName}': ${fallbackError}`);
+			}
```

Comment:
Why would we do this and not simply fallback to the `shell.openExternal`?

Reply:
Hi @bpasero! My mistake - I think I should have tried to make better use of the existing methods and the `open` library - since from what I now understand, the `open` library knows how to launch the different browser links properly without requiring me to make that manual if, else-if, and else block.

Reply:
Do you by chance, have any further suggestions on how I should attempt to resolve this issue properly? I think I might need to do a little bit more research on the `openExternal` and `openExternalBrowser` methods + the `open` library to fix this while introducing the least amount of change.